java/vector/src/main/java/org/apache/arrow/vector/complex/NullableMapVector.java

```diff
@@ -86,7 +86,8 @@ public NullableMapVector(String name, BufferAllocator allocator, FieldType field
   @Override
   public Field getField() {
     Field f = super.getField();
-    return new Field(f.getName(), true, f.getType(), f.getChildren());
+    FieldType type = new FieldType(true, f.getType(), f.getFieldType().getDictionary(), f.getFieldType().getMetadata());
```

Comment:
Is this change orthogonal to the padding issue? 

Maybe we should have a separate PR or reflect this in the commit message?